test/CPUActivityAgentTest.cpp

```diff
@@ -595,27 +519,27 @@ TEST_F(CPUActivityAgentTest, adjust_platform_signal_out_of_bounds)
     EXPECT_TRUE(m_agent->do_write_batch());
 }
 
-TEST_F(CPUActivityAgentTest, adjust_platform_nan)
+TEST_F(CPUActivityAgentTest, no_mem_constconfig)
 {
-    const std::vector<double> policy_nan(m_num_policy, NAN);
-    std::vector<double> policy;
-    policy = policy_nan;
-    EXPECT_CALL(*m_gov, validate_policy(m_cpu_freq_min, m_cpu_freq_max)).Times(1);
-    m_agent->validate_policy(policy);
+    EXPECT_CALL(*m_platform_io, push_signal("MSR::QM_CTR_SCALED_RATE", _, _)).Times(M_NUM_PACKAGE);
+    EXPECT_CALL(*m_platform_io, push_signal("MSR::CPU_SCALABILITY_RATIO", _, _)).Times(M_NUM_CORE);
+    EXPECT_CALL(*m_platform_io, push_signal("CPU_UNCORE_FREQUENCY_STATUS", _, _)).Times(M_NUM_PACKAGE);
 
-    //Sample
-    std::vector<double> tmp;
-    double mock_active = 0.0;
-    EXPECT_CALL(*m_platform_io, sample(CPU_SCALABILITY_IDX))
-                .WillRepeatedly(Return(mock_active));
-    EXPECT_CALL(*m_platform_io, sample(QM_CTR_SCALED_RATE_IDX))
-                .WillRepeatedly(Return(mock_active));
-    EXPECT_CALL(*m_platform_io, sample(CPU_UNCORE_FREQUENCY_IDX))
-                .WillRepeatedly(Return(m_cpu_uncore_freq_max));
-    m_agent->sample_platform(tmp);
+    EXPECT_CALL(*m_platform_io, push_control("CPU_UNCORE_FREQUENCY_MIN_CONTROL", _, _)).Times(M_NUM_PACKAGE);
+    EXPECT_CALL(*m_platform_io, push_control("CPU_UNCORE_FREQUENCY_MAX_CONTROL", _, _)).Times(M_NUM_PACKAGE);
+
+    EXPECT_CALL(*m_platform_io, write_control("MSR::PQR_ASSOC:RMID", _, _, _)).Times(1);
+    EXPECT_CALL(*m_platform_io, write_control("MSR::QM_EVTSEL:RMID", _, _, _)).Times(1);
+    EXPECT_CALL(*m_platform_io, write_control("MSR::QM_EVTSEL:EVENT_ID", _, _, _)).Times(1);
+    EXPECT_CALL(*m_platform_io, read_signal("CPU_UNCORE_FREQUENCY_MIN_CONTROL", _, _)).Times(1);
+    EXPECT_CALL(*m_platform_io, read_signal("CPU_UNCORE_FREQUENCY_MAX_CONTROL", _, _)).Times(1);
```

Comment:
This is almost ``SetUpDefaultExpectCalls``, so I can't help but think there might be an opportunity to factor out some code. If it ends up being more trouble than any benefit that provides, feel free to disregard.